portal/apps/org.wso2.carbon.iam.portal.common/src/main/fragments/profile/profile.js

```diff
@@ -0,0 +1,223 @@
+            if (cause instanceof java.lang.reflect.InvocationTargetException) {
+                message = cause.getTargetException().message;
+            }
+        }
+    }
+    return {success: false, message: message};
+}
+
+function updateUserProfile(userId, updatedClaims) {
+
+    try {
+        var profileUIEntries = callOSGiService("org.wso2.is.portal.user.client.api.IdentityStoreClientService",
+            "updateUserProfile", [userId, updatedClaims]);
+        return {success: true, message: "Your profile is updated."};
+    } catch (e) {
+        var message = e.message;
+        var cause = e.getCause();
+        if (cause !== null) {
+            //the exceptions thrown by the actual osgi service method is wrapped inside a InvocationTargetException.
+            if (cause instanceof java.lang.reflect.InvocationTargetException) {
+                message = cause.getTargetException().message;
+            }
+        }
+    }
+    return {success: false, message: message};
+}
+
+function uploadFile(env, session) {
+
+    try {
+        var Paths = Java.type('java.nio.file.Paths');
+        var System = Java.type('java.lang.System');
+        var Files = Java.type('java.nio.file.Files');
+        var StandardCopyOption = Java.type('java.nio.file.StandardCopyOption');
+
+        var uploadedFile = env.request.files["image"];
+        var imageDirPath = Paths.get(System.getProperty('user.dir'), "images");
+        if (!Files.exists(imageDirPath)) {
+            Files.createDirectories(imageDirPath);
+        }
+
+        var fileName = session.getUser().getUserId();
+        var destination = Paths.get(imageDirPath).resolve(fileName);
+        var sourcePath = Paths.get(uploadedFile.path);
+        var destinationPath = Paths.get(destination);
+
+        Files.copy(sourcePath, destinationPath, StandardCopyOption.REPLACE_EXISTING);
+        return {success: true, message: "Profile image successfully updated"};
+    } catch (e) {
+        var message = e.message;
+        var cause = e.getCause();
+        if (cause !== null) {
+            //the exceptions thrown by the actual osgi service method is wrapped inside a InvocationTargetException.
+            if (cause instanceof java.lang.reflect.InvocationTargetException) {
+                message = cause.getTargetException().message;
+            }
+        }
+    }
+    return {success: false, message: message};
+}
+
+function isProfileImageAvailbale(session) {
+    var usernameChar = session.getUser().getUsername().charAt(0);
+
+    var Paths = Java.type('java.nio.file.Paths');
+    var System = Java.type('java.lang.System');
+    var Files = Java.type('java.nio.file.Files');
+    var File = Java.type('java.io.File');
+    var imageDirPath = Paths.get(System.getProperty('user.dir'), "images");
+    if (!Files.exists(imageDirPath)) {
+        return {profileImage: false, usernameChar: usernameChar};
+    }
+    else {
+        var file = new File(imageDirPath.toString());
```

Comment:
+1. Don't use the old API. You can do Path path = Paths.get(imageDirPath.toString());